picky-asn1-x509/src/pkcs7/crls.rs

```diff
@@ -0,0 +1,245 @@
+    {
+        match &self {
+            RevocationInfoChoice::Crl(certificate_list) => certificate_list.serialize(serializer),
+            RevocationInfoChoice::Other(other_revocation_info_format) => {
+                other_revocation_info_format.serialize(serializer)
+            }
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for RevocationInfoChoice {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as de::Deserializer<'de>>::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        use std::fmt;
+
+        struct Visitor;
+
+        impl<'de> de::Visitor<'de> for Visitor {
+            type Value = RevocationInfoChoice;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a valid DER-encoded RevocationInfoChoice")
+            }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::SeqAccess<'de>,
+            {
+                let tag_peeker: TagPeeker = seq_next_element!(seq, RevocationInfoChoice, "choice tag");
+
+                let revocation_info_choice = match tag_peeker.next_tag {
+                    Tag::CTX_1 => RevocationInfoChoice::Other(seq_next_element!(
+                        seq,
+                        RevocationInfoChoice,
+                        "OtherRevocationInfoFormat "
+                    )),
+                    _ => RevocationInfoChoice::Crl(seq_next_element!(seq, RevocationInfoChoice, "CertificateList")),
+                };
+
+                Ok(revocation_info_choice)
+            }
+        }
+
+        deserializer.deserialize_enum("RevocationInfoChoice", &["Crl", "Other"], Visitor)
+    }
+}
+
+/// CRLs are specified in X.509
+/// ``` not_rust
+/// [RFC 5280 #5.1](https://datatracker.ietf.org/doc/html/rfc5280#section-5.1)
+/// CertificateList  ::=  SEQUENCE  {
+///         tbsCertList          TBSCertList,
+///         signatureAlgorithm   AlgorithmIdentifier,
+///         signatureValue       BIT STRING  }
+///
+///  TBSCertList  ::=  SEQUENCE  {
+///         version                 Version OPTIONAL,
+///                                      -- if present, MUST be v2
+///         signature               AlgorithmIdentifier,
+///         issuer                  Name,
+///         thisUpdate              Time,
+///         nextUpdate              Time OPTIONAL,
+///         revokedCertificates     SEQUENCE OF SEQUENCE  {
+///              userCertificate         CertificateSerialNumber,
+///              revocationDate          Time,
+///              crlEntryExtensions      Extensions OPTIONAL
+///                                       -- if present, version MUST be v2
+///                                   }  OPTIONAL,
+///         crlExtensions           [0]  EXPLICIT Extensions OPTIONAL
+///                                       -- if present, version MUST be v2
+///                                 }
+/// ```
```

Comment:
Sorry, maybe I missed in the first review, but this one should be in the docstring of `TBSCertList`